lib/Container/Compiler/FieldType/RichTextIndexablePass.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Netgen\EzPlatformSearchExtra\Container\Compiler\FieldType;
+
+use Netgen\EzPlatformSearchExtra\Core\FieldType\XmlText\Indexable as IndexableXmlText;
+use RuntimeException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class RichTextIndexablePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $configurationParameterName = 'netgen_ez_platform_search_extra_configuration';
+
+        if (!$container->hasParameter($configurationParameterName)) {
+            return;
+        }
+
+        $configuration = $container->getParameter($configurationParameterName);
+
+        if (!isset($configuration['indexable_field_type']['ezrichtext'])) {
+            return;
+        }
+
+        $configuration = $configuration['indexable_field_type']['ezrichtext'];
+
+        if ($configuration['override'] === true) {
+            $this->redefineIndexableImplementation($container, $configuration);
+        }
+    }
+
+    private function redefineIndexableImplementation(ContainerBuilder $container, array $configuration)
+    {
+        try {
+            $originalServiceId = $this->getOriginalServiceId($container);
+        } catch (RuntimeException $e) {
+            return;
+        }
+
+        $definition = $container->findDefinition($originalServiceId);
+
+        $definition->setClass(IndexableXmlText::class);
```

Comment:
Have you considered using decorators (https://symfony.com/doc/current/service_container/service_decoration.html) here, instead of redefining the service?

Reply:
Yes, but it doesn't work from compiler pass, probably because the services are already registered in another compiler pass.

Reply:
Really? Didn't test, but it definitelly sounds weird.